### PR TITLE
Add support for `--base` CLI argument

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Astro currently allows `--site` and `--root` and a few other CLI arguments.
- This PR adds support for `--base` as well
- Why? Sometimes CLI arguments are very handy, like if you're building an official GitHub Action ([`withastro/action`](https://github.com/withastro/action)) and want to automatically pass in the `--site` and `--base` that GitHub already knows.

See also https://github.com/actions/configure-pages/issues/30

## Testing

Tested manually because we don't have equivalent tests for `--site` or other arguments

## Docs

https://github.com/withastro/docs/pull/1646
/cc @withastro/maintainers-docs

### CODE REVIEW

**Review by tony-sull — APPROVED** [review]
Nice!  Might be worth adding a CLI log that could be checked by a test as well *if* the CLI tests are in a decent spot for it, but not 100% necessary 👍

**Comment by sarah11918** [comment]
Removing the docs-maintainers review, as the changes to `@types/astro.ts` won't affect docs!

**Comment by jasikpark** [comment]
can you update astro help for these?
```
❯ npx astro build --help

   astro  v1.4.0 Build faster websites.

  astro [command] [...flags]

   Commands 
              add  Add an integration.
            build  Build your project and write it to disk.
            check  Check your project for errors.
              dev  Start the development server.
             docs  Open documentation in your web browser.
          preview  Preview your build locally.
        telemetry  Configure telemetry settings.

   Global Flags 
  --config <path>  Specify your config file.
    --root <path>  Specify your project root folder.
        --verbose  Enable verbose logging.
         --silent  Disable all logging.
        --version  Show the version number and exit.
           --help  Show this help message.


```

interesting that `astro build -h` doesn't print help & just builds

**Comment by natemoo-re** [comment]
Thanks @jasikpark! Updated the `help` command output. Would you mind opening a new issue for tracking `astro build --help`?

**Comment by jasikpark** [comment]
does this classify as a bug report? since that's the only type of issue in "new issue" that creates an actual issue?

**Comment by jasikpark** [comment]
> Thanks @jasikpark! Updated the `help` command output. Would you mind opening a new issue for tracking `astro build --help`?

https://github.com/withastro/astro/issues/4935

File: .changeset/wise-swans-live.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
```
Comment (matthewp) [inline .changeset/wise-swans-live.md:2]: Seems like this should be a `minor`

**Review by matthewp — CHANGES_REQUESTED** [review]
Think this is a minor change.

**Review by matthewp — APPROVED** [review]
lgtm for next minor

**Comment by matthewp** [comment]
@natemoo-re can you fill out the changeset a bit more? Something like a paragraph, maybe a code example, so that this could be copy/pasted into a blog post. Thank you!

**Pull request merged**